server/datastore/mysql/labels.go

```diff
@@ -226,7 +226,24 @@ func (d *Datastore) SaveLabel(ctx context.Context, label *fleet.Label) (*fleet.L
 
 // DeleteLabel deletes a fleet.Label
 func (d *Datastore) DeleteLabel(ctx context.Context, name string) error {
-	return d.deleteEntityByName(ctx, labelsTable, name)
+	return d.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		var labelID uint
+		err := d.writer.GetContext(ctx, &labelID, `select id FROM labels WHERE name = ?`, name)
```

Comment:
Should use `tx`? (same below)

Reply:
Mmmman, I wonder if we can encode this into a lint rule.